packages/cli/src/routeGeneration/templates/express.hbs

```diff
@@ -72,7 +72,14 @@ export function RegisterRoutes(app: express.Express) {
             }
 
             {{#if ../../iocModule}}
-            const controller: any = iocContainer.get<{{../name}}>({{../name}});
+            function isFunction(obj: any): obj is Function { return typeof obj === 'function'; }
+            interface Container<T> { get<T>(type: any): any; }
+
+            let container: any = iocContainer;
+            if (isFunction(iocContainer)) {
+                container = iocContainer(request);
+            }
+            const controller: any = (container as Container<{{../name}}> ).get<{{../name}}>({{../name}});
```

Comment:
```suggestion
            let container: any = typeof iocContainer === 'function' ? iocContainer(request) : iocContainer;
            const controller: any = container.get<{{../name}}>({{../name}});
```

Reply:
I think I can drop the `if` and `isFunction`, but without the `Container` interface there's still the 'Untyped function calls may not accept type arguments.' error.
```ts
interface Container<T> { get<T>(type: any): any; }

const container: any = typeof iocContainer === 'function' ? iocContainer(request) : iocContainer;
const controller: any = (container as Container<{{../name}}>).get<{{../name}}>({{../name}});
```

Reply:
```
fixtures/inversify/routes.ts:572:73 - error TS2349: This expression is not callable.
  Type 'never' has no call signatures.

572      const container: any = typeof iocContainer === 'function' ? iocContainer(request) : iocContainer;
```

I can get this to work by adding in the function signature in the case that it's not a function.

```ts
const container: any = typeof iocContainer === 'function' ?
    (iocContainer as (req:any) => any)(request) : iocContainer;
```